### PR TITLE
nrf_802154_sl: Remove FEM support in 802.15.4 SL

### PR DESCRIPTION
**Pull request by JakubPegza**
FEM support has been moved to MPSL, obsolete implementation
and API has been removed from this repository

Signed-off-by: Pegza, Jakub <Jakub.Pegza@nordicsemi.no>

### CODE REVIEW

**Review by czeslawmakarski — APPROVED** [review]

File: nrf_802154_sl/CHANGELOG.rst [hunk 1]
```diff
@@ -3,12 +3,19 @@
 Changelog
 #########
 
-.. contents::
```
Comment (hubertmis) [inline nrf_802154_sl/CHANGELOG.rst:6]: Should this part be removed? I thought it was added to unify with other libraries.

**Review by hubertmis — CHANGES_REQUESTED** [review]
Good change. Especially I like reduction of size of the libraries.
I'm only concerned about making documentation files inconsistent with other modules in this repository.

**Review by hubertmis — APPROVED** [review]
:+1:

**Comment by hubertmis** [comment]
@FrancescoSer , could you have a look at doc changes in this PR?

**Review by FrancescoSer — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**